### ISSUE TITLE
ECK 2.8.0: enterprise license update fails

### ISSUE THREAD

**Issue by pbdger** [elastic/cloud-on-k8s#7127]
## Bug Report

We have a productive ECK running with license enterprise. This license expires on 2023-08-31. We want to renew the license.

This update fails with an error shown in the output of the operator.

**What did you do?**

We add the new license as described in https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-licensing.html.

**What did you expect to see?**

We expect a renewed license valid until 2025-08-31. 

```kubectl -n eck get configmap elastic-licensing -o json | jq .data```should show the new license data.

**What did you see instead? Under which circumstances?**

```kubectl -n eck get configmap elastic-licensing -o json | jq .data``` returns a basic license as the new license is not accepted.

**Environment**

* ECK version:
	2.8.0

* Kubernetes information:
	* Cloud: Ionos
	* k8s Version 1.26.5
	
* Logs:
```
{
  "log.level": "error",
  "@timestamp": "2023-08-25T12:13:36.090Z",
  "log.logger": "resource-reporter",
  "message": "Failed signature check",
  "service.version": "2.8.0+3940cf4d",
  "service.type": "eck",
  "ecs.version": "1.4.0",
  "error": "crypto/rsa: verification error",
  "error.stack_trace": "github.com/elastic/cloud-on-k8s/v2/pkg/controller/common/license.(*Verifier).Valid
\t/go/src/github.com/elastic/cloud-on-k8s/pkg/controller/common/license/verifier.go:38
github.com/elastic/cloud-on-k8s/v2/pkg/controller/common/license.(*checker).Valid
\t/go/src/github.com/elastic/cloud-on-k8s/pkg/controller/common/license/check.go:115
github.com/elastic/cloud-on-k8s/v2/pkg/controller/common/license.(*checker).CurrentEnterpriseLicense
\t/go/src/github.com/elastic/cloud-on-k8s/pkg/controller/common/license/check.go:81
github.com/elastic/cloud-on-k8s/v2/pkg/license.LicensingResolver.getOperatorLicense
\t/go/src/github.com/elastic/cloud-on-k8s/pkg/license/license.go:164
github.com/elastic/cloud-on-k8s/v2/pkg/license.LicensingResolver.ToInfo
\t/go/src/github.com/elastic/cloud-on-k8s/pkg/license/license.go:93
github.com/elastic/cloud-on-k8s/v2/pkg/license.ResourceReporter.Get
\t/go/src/github.com/elastic/cloud-on-k8s/pkg/license/reporter.go:85
github.com/elastic/cloud-on-k8s/v2/pkg/license.ResourceReporter.Report
\t/go/src/github.com/elastic/cloud-on-k8s/pkg/license/reporter.go:67
github.com/elastic/cloud-on-k8s/v2/pkg/license.ResourceReporter.Start
\t/go/src/github.com/elastic/cloud-on-k8s/pkg/license/reporter.go:55
github.com/elastic/cloud-on-k8s/v2/cmd/manager.asyncTasks.func1
\t/go/src/github.com/elastic/cloud-on-k8s/cmd/manager/main.go:769"
```

**Comment by thbkrkr** [comment]
Hello, this error is most often the result of an incorrectly formatted License Secret. You should double check it.

**Comment by BenB196** [comment]
Hi @pbdger, I've run into a similar issue in the past, the issue I had was I download the Elastic Stack license, but you want to make sure you download the Orchestration license as they are slightly different formats.

**Comment by pbdger** [comment]
You are right, @BenB196.

There is no property in the license file that describes the type of license in more detail. The error displayed in this case is not meaningful.
Maybe this can be improved in a future version and a better message can be displayed.

**Comment by rauanmayemir** [comment]
This is still happening. Worked the first time, but then patching the secret with new license file next year resulted in the same error.

**Comment by pebrc** [comment]
@rauanmayemir thanks for the follow-up. The most likely cause on renewal is the same as the original issue: accidentally downloading the **Elastic Stack license** instead of the **Orchestration license** from the Elastic portal. These are signed with different RSA keys, and ECK only accepts the Orchestration variant — but the error message (`crypto/rsa: verification error`) gives no indication of this.

I've filed https://github.com/elastic/cloud-on-k8s/issues/9217 to improve the error message and investigate using the key fingerprint embedded in the signature to give a more actionable diagnostic when the wrong license type is used.

**Comment by pebrc** [comment]
(Correction: the tracking issue is https://github.com/elastic/cloud-on-k8s/issues/9261, not #9217 as I mentioned above.)

**Comment by rauanmayemir** [comment]
@pebrc yeah, this turned out to be exactly the case. using the correct license file resolved the issue.
